2025-07-09 15:48:39,677 [WARNING] PyNaCl is not installed, voice will NOT be supported
2025-07-09 15:48:39,677 [INFO] logging in using static token
2025-07-09 15:48:40,557 [INFO] Shard ID None has connected to Gateway (Session ID: 80053070f8e5c525f203081ae68b8c7b).
2025-07-09 15:48:50,226 [ERROR] Ignoring exception in on_message
Traceback (most recent call last):
  File "/home/justin/Infosys/discord_chatbot/.venv/lib/python3.12/site-packages/discord/client.py", line 481, in _run_event
    await coro(*args, **kwargs)
  File "/home/justin/Infosys/discord_chatbot/discord_bot.py", line 164, in on_message
    response = agent_graph.invoke({
               ^^^^^^^^^^^^^^^^^^^^
  File "/home/justin/Infosys/discord_chatbot/.venv/lib/python3.12/site-packages/langgraph/pregel/__init__.py", line 2719, in invoke
    for chunk in self.stream(
  File "/home/justin/Infosys/discord_chatbot/.venv/lib/python3.12/site-packages/langgraph/pregel/__init__.py", line 2436, in stream
    for _ in runner.tick(
  File "/home/justin/Infosys/discord_chatbot/agent.py", line 247, in conductor
    memory_storage.add_message(
  File "/home/justin/Infosys/discord_chatbot/memory_storage.py", line 25, in add_message
    if len(self.local_memory.total_chat_history.get(channel_id)) > 20:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: object of type 'InMemoryVectorStore' has no len()
During task with name 'conductor' and id 'eba5387d-a3bc-8929-a88f-1cf87890d84d'
